### PR TITLE
Support charmstore url and version and jem url and version config.

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/83/
Test FAILed.

**Comment by jaycee** [comment]
QA:

Make a new package from an up to date checkout of the develop branch of juju gui and copy it over the one in releases.
Deploy the charm to a model of your choice.
Confirm the defaults work--charm browsing etc should work and jem should not be connected.
Update the charmstore url and jem url to point at staging. jem should attempt to connect as there is an env there shared to yellow. It should load properly. Charmbrowsing should work.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/84/
Test PASSed.

File: config.yaml [hunk 1]
```diff
@@ -116,6 +116,11 @@ options:
       charmworld-url option above.
     type: string
     default: https://api.jujucharms.com/charmstore/
+  charmstore-version:
+    description: |
+      The version of the charmstore api to use.
```
Comment (bac) [inline config.yaml]: s/api/API/

File: config.yaml [hunk 2]
```diff
@@ -170,6 +175,10 @@ options:
     type: string
     default: ''
     description: Publicly accessible URL of a Juju Environment Manager.
+  jem-version:
+    type: string
+    default: 'v1'
```
Comment (bac) [inline config.yaml]: You used quotes here but not for charmstore-version. I don't think they are needed.

File: config.yaml [hunk 3]
```diff
@@ -116,6 +116,11 @@ options:
       charmworld-url option above.
     type: string
     default: https://api.jujucharms.com/charmstore/
+  charmstore-version:
+    description: |
+      The version of the charmstore api to use.
+    type: string
+    default: v4
```
Comment (hatched) [inline config.yaml]: Do you have to quote this? You're quoting the similar value below for JEM

**Comment by bac** [comment]
:+1: after making the use of quotes consistent in config.yaml.  Thanks.

File: config.yaml (hunk 3 above)
Comment (jaycee) [inline config.yaml]: No, the other value didn't have to be quoted.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/85/
Test PASSed.

**Comment by hatched** [comment]
:+1: QA :ok: Thanks this is a nice improvement!

**Comment by jaycee** [comment]
Thanks @bac and @hatched.
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**